test/puppetlabs/puppetdb/http/facts_test.clj

```diff
@@ -581,73 +575,74 @@
-                  (if (= endpoint "/v4/environments/DEV/facts")
-                    (do
-                      (is (= {:error "No information is known about environment DEV"} (json/parse-string body true)))
-                      (is (= status http/status-not-found)))
-                    (do
-                      (is (empty? (json/parse-stream (io/reader body) true)))
-                      (is (= status http/status-ok))))))
-
-              (testing "config with only a single database returns results"
-                (let [request (get-request endpoint (json/parse-string nil))
-                      {:keys [status body headers]} (one-db-app request)]
-                  (is (= status http/status-ok))
-                  (is (= (headers "Content-Type") c-t))
-                  (is (= [{:certname "foo1" :name "domain" :value "testing.com" :environment "DEV"}
-                          {:certname "foo1" :name "hostname" :value "foo1" :environment "DEV"}
-                          {:certname "foo1" :name "kernel" :value "Linux" :environment "DEV"}
-                          {:certname "foo1" :name "operatingsystem" :value "Debian" :environment "DEV"}
-                          {:certname "foo1" :name "some_version" :value "1.3.7+build.11.e0f985a" :environment "DEV"}
-                          {:certname "foo1" :name "uptime_seconds" :value "4000" :environment "DEV"}]
-                         (sort-by :name (json/parse-stream (io/reader body) true))))))))))))
+        read-config (-> read-db-config defaulted-read-db-config)
+        write-config (-> write-db-config defaulted-write-db-config)]
+    (clear-db-for-testing! read-config)
+    (clear-db-for-testing! write-config)
+    (init-db read-config true)
+    (init-db write-config false)
+    (svc-utils/call-with-puppetdb-instance
+     config
+     (fn []
+       (let [pdb (get-service svc-utils/*server* :PuppetDBServer)
+             shared-globals (cli-svc/shared-globals pdb)
+             read-db (:scf-read-db shared-globals)
+             write-db (:scf-write-db shared-globals)
+             one-db-app (test-app write-db)
+             two-db-app (test-app read-db write-db)
+             facts1 {"domain" "testing.com"
+                     "hostname" "foo1"
+                     "kernel" "Linux"
+                     "operatingsystem" "Debian"
+                     "some_version" "1.3.7+build.11.e0f985a"
+                     "uptime_seconds" "4000"}]
+
+         (jdbc/with-transacted-connection write-db
+           (scf-store/add-certname! "foo1")
+           (scf-store/add-facts! {:certname "foo1"
+                                  :values facts1
+                                  :timestamp (now)
+                                  :environment "DEV"
+                                  :producer_timestamp (now)}))
+
+         (testing "queries only use the read database"
+           (let [request (get-request endpoint (json/parse-string nil))
```

Comment:
prexisting, but I think these should all work fine with a single endpoint argument